src/AspNetCoreModuleV2/AspNetCore/Inc/aspnetcore_shim_config.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#pragma once
+#define CS_ASPNETCORE_SECTION                            L"system.webServer/aspNetCore"
+#define CS_ASPNETCORE_PROCESS_EXE_PATH                   L"processPath"
+#define CS_ASPNETCORE_PROCESS_ARGUMENTS                  L"arguments"
+#define CS_ASPNETCORE_HOSTING_MODEL                      L"hostingModel"
+
+enum APP_HOSTING_MODEL
+{
+    HOSTING_UNKNOWN = 0,
+    HOSTING_IN_PROCESS,
+    HOSTING_OUT_PROCESS
+};
+
+class ASPNETCORE_SHIM_CONFIG : IHttpStoredContext
+{
+public:
+    virtual
+    ~ASPNETCORE_SHIM_CONFIG();
+
+    static
+    HRESULT
+    GetConfig(
+        _In_  IHttpServer             *pHttpServer,
```

Comment:
I think we can get ride of this method as now shim_config is owned by applicationinfo object. Can be done in next PR